src/Surfnet/ServiceProviderDashboard/Domain/Entity/Entity/Protocol.php

```diff
@@ -77,4 +77,9 @@ public function asArray(): array
             'type' => $this->getProtocol(),
         ];
     }
+
+    public function getManagedProtocol(): string
+    {
+        return array_search($this->protocol, self::$protocolMapping);
+    }
```

Comment:
Nope, did not see any yellow swirly lines. I guess this is because the array_search returns a mixed value of integer|string|false. So the function covers the string, but not the other types. I will make the function more verbose and implement your suggestion, together with the exception.